src/script/page/Modals/UserModal/UserModal.tsx

```diff
@@ -0,0 +1,184 @@
+  const replaceLinkLegalHold = replaceLink(Config.getConfig().URL.SUPPORT.LEGAL_HOLD_BLOCK, '', 'read-more-legal-hold');
+
+  useEffect(() => {
+    if (userId) {
+      userRepository
+        .getUserById(userId)
+        .then(user => {
+          if (user.isDeleted) {
+            return setUserNotFound(true);
+          }
+          setUser(user);
+        })
+        .catch(() => setUserNotFound(true));
+      setIsShown(true);
+    }
+
+    return () => {
+      setUser(null);
+      setUserNotFound(false);
+    };
+  }, [userId.id, userId.domain]);
+
+  return (
+    <div className="user-modal">
+      <ModalComponent
+        isShown={isShown}
+        onBgClick={hide}
+        onClosed={onModalClosed}
+        data-uie-name={user ? 'modal-user-profile' : userNotFound ? 'modal-cannot-open-profile' : ''}
+      >
+        <div className="modal__header">
+          {userNotFound && (
+            <h2 className="modal__header__title" data-uie-name="status-modal-title">
+              {t('userNotFoundTitle', brandName)}
+            </h2>
+          )}
+          <Icon.Close className="modal__header__button" onClick={hide} data-uie-name="do-close" />
+        </div>
+        <div className={cx('modal__body user-modal__wrapper', {'user-modal__wrapper--max': !user && !userNotFound})}>
+          {user && (
+            <>
+              <UserDetails participant={user} isSelfVerified={isSelfVerified} classifiedDomains={classifiedDomains} />
+              <EnrichedFields user={user} showDomain={isFederated} />
+              {user.isBlockedLegalHold() && (
+                <div className="modal__message" data-uie-name="status-blocked-legal-hold">
+                  {t('modalUserBlockedForLegalHold', {}, replaceLinkLegalHold)}
+                </div>
+              )}
+              {user.isBlockedLegalHold() && (
+                <UserActions
+                  user={user}
+                  actionsViewModel={actionsViewModel}
+                  onAction={hide}
+                  isSelfActivated={isActivatedAccount}
+                  selfUser={self}
+                />
+              )}
+            </>
+          )}
+          {isShown && !user && !userNotFound && (
+            <div className="loading-wrapper">
+              <Icon.Loading aria-hidden="true" />
+            </div>
+          )}
+          {userNotFound && (
+            <>
+              <div className="modal__message" data-uie-name="status-modal-text">
+                {t('userNotFoundMessage', brandName)}
+              </div>
+              <div className="modal__buttons">
+                <div className="modal__button modal__button--confirm" data-uie-name="do-ok" onClick={hide}>
+                  {t('modalAcknowledgeAction')}
+                </div>
+              </div>
+            </>
+          )}
+        </div>
+      </ModalComponent>
+    </div>
+  );
+};
+
+export default UserModalComponent;
```

Comment:
Maybe we do not need to export the component itself, Only the API is needed

Reply:
needed for testing